domains/client/domain-executor/src/domain_worker.rs

```diff
@@ -333,13 +333,20 @@ where
         .runtime_api()
         .extrinsics_shuffling_seed(&block_id, header)?;
 
-    let core_bundles = primary_chain_client
-        .runtime_api()
-        .extract_core_bundles(&block_id, extrinsics, domain_id)?;
+    let domain_bundles = if domain_id.is_system() {
+        todo!("Migrate system_domain_worker to be based on domain_worker")
+    } else if domain_id.is_core() {
+        let core_bundles = primary_chain_client
+            .runtime_api()
+            .extract_core_bundles(&block_id, extrinsics, domain_id)?;
+        DomainBundles::Core(core_bundles)
+    } else {
+        unreachable!("Open domains are unsupported")
```

Comment:
Is this something that user can provide and cause client to panic? If so it needs to be handled gracefully or else I'd like to see explanation why it is not possible in the comment.

Reply:
No, a user can't influence it as the validity of domain id is checked before actually starting the domain node. Currently, only the core domains are implemented/supported, and open domains are not supported, this branch here is more for completeness here. Unsure if changing it to `unimplemented!()` can make it clearer.